### PR TITLE
[ian] Project 2: Dice - Beat that!

### PR DESCRIPTION
**Pull request by iannyip**
### Submission 1: Base Case

Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

**How many hours did you spend on this assignment?**
40 min

**Please fill in one error and/or error message you received while working on this assignment.**
extra parenthesis in condition

**What part of the assignment did you spend the most time on?**
Figuring out the condition for input validation (ensure that user inputs only 1 or 2)
Case where user inputs ((input != 1 AND input != 2 ) OR input == "")

**Comfort Level (1-5):**
5

**Completeness Level (1-5):**
5

**What did you think of this deliverable?**
Manageable!

**Is there anything in this code that you feel pleased about?**
Completion

### CODE REVIEW

**Comment by iannyip** [comment]
### Submission 2: More comfortable (variable number of dice)
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

**How many hours did you spend on this assignment?**
40 min

**Please fill in one error and/or error message you received while working on this assignment.**
accidentally console.log('p1Combined: ' = orderedResult);
correct: console.log('p1Combined: ' + orderedResult);

**What part of the assignment did you spend the most time on?**
splitting the user's numeric input into an array

**Comfort Level (1-5):**
4

**Completeness Level (1-5):**
5

**What did you think of this deliverable?**
Manageable!

**Is there anything in this code that you feel pleased about?**
Completion

File: script.js [hunk 1]
```diff
@@ -1,4 +1,129 @@
+// Variable declaration
+var gameMode = 'select dice number'
+var numberOfDice = 0;
+var p1Rolls = [];
+var p2Rolls = [];
+var p1combinedNumber = 0;
+var p2combinedNumber = 0;
+
+// Function 1: randomizer between 1-6
+var diceRoll = function () {
```
Comment (kai-rocket) [inline script.js:10]: Reminder to name functions starting with a verb if possible for clarity

File: script.js [hunk 2]
```diff
@@ -1,4 +1,129 @@
...
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal);
+  var diceNumber = randomInteger + 1;
+  return diceNumber;
+};
+
+// Function 2: rolls dice X number of times
+var rollsInArray = function(diceCount){
```
Comment (kai-rocket) [inline script.js:18]: Same, function name can start with verb for clarity

File: script.js [hunk 3]
```diff
@@ -1,4 +1,129 @@
...
+  var thisArray = [];
+  while (counter < diceCount){
+    thisArray.push(diceRoll());
+    counter = counter + 1;
+  }
+  return thisArray;
+}
+
+// Function 3: print the roll result
+var printRollResult = function(playerArray){
```
Comment (kai-rocket) [inline script.js:29]: Reminder to code with ESLint so your code format is always standardised!

File: script.js [hunk 4]
```diff
@@ -1,4 +1,129 @@
...
+  var counter = 0;
+  var thisArray = [];
+  while (counter < diceCount){
+    thisArray.push(diceRoll());
+    counter = counter + 1;
+  }
+  return thisArray;
+}
+
+// Function 3: print the roll result
```
Comment (kai-rocket) [inline script.js:28]: Great idea to make a function for the output string to avoid repetition!

File: script.js [hunk 5]
```diff
@@ -1,4 +1,129 @@
...
+    secondLine = secondLine + ' ' + playerArray[counter] ;
+    thirdLine = thirdLine + ' ' +counter;
+    counter = counter + 1;
+  }
+  msgResult = firstLine + secondLine + thirdLine + fourthLine
+  return msgResult; 
+}
+
+// Function 4: Creates number from number input
+var numberOrdered = function (userInputOrder, userDice){
```
Comment (kai-rocket) [inline script.js:47]: This function name is confusing- would recommend a clearer name!

Comment (kai-rocket) [reply script.js:47]: `userInputOrder` is relatively clear, `userDice` isn't super clear

Comment (kai-rocket) [reply script.js:47]: Would recommend trying JSDoc comment format for clearer function-level commenting https://jsdoc.app/about-getting-started.html

File: script.js [hunk 6]
```diff
@@ -1,4 +1,129 @@
...
+  var positions = userInputOrder.toString(10).split("").map(Number);
```
Comment (kai-rocket) [inline script.js:48]: Nice use of `split` and `map`

File: script.js [hunk 7]
```diff
@@ -1,4 +1,129 @@
...
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = '';
+  console.log('Game Mode: ' + gameMode);
+
+  // Game Mode 1: User selects number of dice
+  if (gameMode == 'select dice number'){
+    console.log(input);
+    console.log(isNaN(Number(input)));
+    if (isNaN(Number(input)) || input == "" ){
```
Comment (kai-rocket) [inline script.js:72]: Great validation

File: script.js [hunk 8]
```diff
@@ -1,4 +1,129 @@
...
+    if (isNaN(Number(input)) || input == "" || p1Rolls.length != input.toString(10).split("").map(Number).length) {
+      myOutputValue = 'Player 1, please input a valid number'
+      gameMode = 'p1 choose';
+      return myOutputValue;
+    }
+    p1combinedNumber = numberOrdered(input, p1Rolls);
+    myOutputValue = 'Player 1, you chose order ' + input + '. <br> Your number is ' + p1combinedNumber + '.<br> It is now player 2\'s turn <br><br> Player 2, click submit.';
+  }
+  // Game Mode 4: Player 2 rolls
+  else if (gameMode == 'p2 roll'){
```
Comment (kai-rocket) [inline script.js:99]: Some of these else ifs may not be necessary if we return from each conditional

**Review by kai-rocket — COMMENTED** [review]
All in all- great work!!! Code structure makes sense and great work getting more comfortable to work! Can improve on naming and simplifying code logic in `main`. See more detailed comments inline

**Pull request closed without merging**